src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -436,6 +437,16 @@ private async Task StopAsyncCore(bool disposing)
             }
         }
 
+#if NETCOREAPP3_0
+        public async IAsyncEnumerable<object> StreamAsyncCore(string methodName, Type returnType, object[] args, CancellationToken cancellationToken = default)
+        {
+            var stream = (await StreamAsChannelCoreAsyncCore(methodName, returnType, args, cancellationToken)).ReadAllAsync();
+            await foreach(var item in AsyncEnumerableAdapters.MakeCancelableAsyncEnumerable(stream, cancellationToken)) {
```

Comment:
You shouldn't need the `await foreach` here anymore right? `MakeCancelableAsyncEnumerable` just returns the think we need. It can just be like this:

```suggestion
            return AsyncEnumerableAdapters.MakeCancelableAsyncEnumerable(stream, cancellationToken);
```

Reply:
Ah wait, I see why. Because that would end up being a `Task<IAsyncEnumerable<T>>` when we want an `IAsyncEnumerable<T>`. Unfortunate, but makes sense. Disregard!

Reply:
If you are going to be doing this a lot (creating IAsyncEnumerable\<T> from async methods), you could consider creating a helper like MakeCancelableAsyncEnumerableFromChannel but that takes a Task\<Channel\<T>> and handles it to an IAsyncEnumerable\<T> that defers the await until the first call to MoveNextAsync. Then you can return that directly without creating an iterator method. 

The only observable difference would be that any exception that awaiting StreamAsChannelCoreAsyncCore would cause will be deferred.

This is what we do with ExecuteDataReaderAsync.

Reply:
That's a good idea. Probably worth making a new issue to track it. We could convert `Task<ChannelReader<T>>` to `ChannelReader<T>` in the same way.

Reply:
I think that's something we can look at if we encounter an issue. Let's get this rolling now and see how it lands :).

Reply:
FWIW, one correction to what I said before: You want to pass a Func<Task<Channel<T>> to allow deferring the call (in this case to StreamAsChannelCoreAsyncCore) completely instead of passing an already initiated Task<T>.

> Probably worth making a new issue to track it.

Sure. If you see value. I am back to lurker mode :smile: